src/app/core/database/mock-database.ts

```diff
@@ -15,431 +15,51 @@
-   * @return string of the padded input
-   */
-  private zeroPad(str: string | number, length: number = 14): string {
-    // with ECMAScript 2017 you can do a one-liner: 'return str.toString().padStart(length, '0');'
-    let res = str.toString();
-    while (res.length < length) {
-      res = "0" + res;
-    }
-    return res;
+  static createWithInMemoryDB(
+    dbname: string = "in-memory-mock-database",
+    loggingService: LoggingService = new LoggingService()
+  ): MockDatabase {
+    PouchDB.plugin(memory).plugin(mapreduce);
+    return new MockDatabase(
+      new PouchDB(dbname, { adapter: "memory" }),
+      loggingService
+    );
   }
 
-  private isWithinLastMonths(
-    date: Date,
-    now: Date,
-    numberOfMonths: number
-  ): boolean {
-    if (!date) {
-      return false;
-    }
-    date = new Date(date);
+  private indexPromises: Promise<any>[] = [];
 
-    let months;
-    months = (now.getFullYear() - date.getFullYear()) * 12;
-    months -= date.getMonth();
-    months += now.getMonth();
-    if (months < 0) {
-      return false;
-    }
-    return months <= numberOfMonths;
-  }
-
-  private getStatsReduceFun(
-    keyFun: (any) => string,
-    valueFun: (any) => number
-  ) {
-    return (acc, value) => {
-      const stats = {
-        key: keyFun(value),
-        value: { count: 1, sum: valueFun(value) },
-      };
-      const existing = acc.filter((x) => x.key === keyFun(value));
-      if (existing.length > 0) {
-        const v = existing[0].value;
-        v.count++;
-        v.sum = v.sum + stats.value.sum;
-      } else {
-        acc.push(stats);
-      }
-
-      return acc;
-    };
+  public saveDatabaseIndex(designDoc: any): Promise<any> {
+    const indexPromise = super.saveDatabaseIndex(designDoc);
+    this.indexPromises.push(indexPromise);
+    return indexPromise;
   }
 
   /**
-   * Currently not implemented for MockDatabase!
-   *
-   * Check (and extend) the `query` method for hard-coded mocks of some specific queries.
-   *
-   * @param designDoc
+   * Returns a promise that will resolve, once all indices are built.
+   * This function should be called AFTER saveDatabaseIndex was called for each index.
+   * If a second index is saved after the first one is done and this function is called immediately after saving the
+   * first index, then the promise will not wait for the second index to be done.
    */
-  saveDatabaseIndex(designDoc) {
-    // TODO: implement mock query
-    /* SAMPLE INPUT:
-    const designDoc = {
-      _id: '_design/attendance_index',
-      views: {
-        by_name: {
-          map: (doc) => { emit(doc.child); }
-        },
-        by_month: {
-          map: (doc) => { emit(doc.month); }
-        }
-      }
-    };
-     */
-
-    // tslint:disable-next-line:no-console
-    console.debug('MockDatabase does not implement "saveDatabaseIndex()"');
-    return Promise.resolve({});
+  public waitForIndexing(): Promise<any> {
```

Comment:
let's put this in the `PouchDatabase` parent class, so we can also wait for completion of its indices in performance tests?